Improve integration with the ganalytics package

jdeboer opened the pull request:
Modified the `google_analytics` function to coerce supplied segments and filters objects to the expected classes so that coercion methods from ganalytics can be invoked to coerce ganalytics segments and filters to their googleAnalyticsR equivalents. For this, the relevant googleAnalyticsR classes also needed to be formally declared and exported. Once these changes are available through CRAN then the ganalytics package can then be updated to import the googleAnalyticsR classes. Note the class `.filter_clauses_ga4` is named as such (with the leading period) as it can be considered to be a virtual class.

MarkEdmondson1234 commented:
I'm getting this error when I try the tests:

```
> google_analytics(ga_id,
+                  date_range = c("2015-07-30","2015-10-01"),
+                  dimensions=c('source','medium'),
+                  metrics = c('sessions','bounces'),
+                  met_filters = fc,
+                  dim_filters = fc2,
+                  filtersExpression = "ga:source!=(direct)")
 Show Traceback
 
 Rerun with Debug
 Error in as(dim_filters, ".filter_clauses_ga4") : 
  no method or default for coercing “dim_fil_ga4” to “.filter_clauses_ga4” 
```

I've not used `as` to set classes before, rather `structure()` or `class()` directly, but let me know if I'm missing something

jdeboer commented:
Hi Mark, sorry about that. I've updated my pull request to set `.filter_clauses_ga4` as a class union of `dim_fil_ga4` and `met_fil_ga4`. That way R will no longer look for a method to coerce a `dim_fil_ga4` object to `.filter_clauses_ga4` because it will already belong to that superclass. I've also performed a test to confirm that this works using the example from the `segment_ga4` help page - which completely executed without errors (except that there are some typos in the example, which I have also updated.)

MarkEdmondson1234 commented:
Thanks!  I need to correct some tests, but the ones concerning this are working
